src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -168,4 +170,19 @@ private BucketEntity findBucketByNameAndCheck(String bucketName) {
         }
         return bucketEntity;
     }
+
+    public List<BucketMetadata> getBucketsByGroups(String sessionId, String ownerName, Optional<String> kind,
+            Optional<String> contentType, Supplier<List<String>> authenticatedUserGroupsSupplier)
+            throws NotAuthenticatedException, AccessDeniedException {
+        List<String> groups;
+
+        if (ownerName == null) {
+            groups = ownerGroupStringHelper.getGroupsWithPrefixFromGroupList(authenticatedUserGroupsSupplier.get());
+            groups.add(BucketService.DEFAULT_BUCKET_OWNER);
+        } else {
+            groups = Collections.singletonList(ownerName);
+        }
```

Comment:
Maybe you could factorize this 'if', cause it is pretty difficult to get your intentions.
Or maybe just put some comment

Reply:
Agree, intentions are difficult to understand there. It's existing code I refactored a bit, but not enough clearly. I ll spend some more time on it until it s more human friendly